File: src/main/java/io/github/nan8279/jhttp_script/script/Console.java
package io.github.nan8279.jhttp_script.script;

import org.mozilla.javascript.Context;

import java.util.HashMap;

/**
 * JavaScript-like console object.
 */
public class Console {
    final private HashMap<String, Integer> counts = new HashMap<>();

    public Console() {
        counts.put("default", 0);
    }

    public void count() {
        counts.replace("default", counts.get("default") + 1);
        System.out.println("default: " + counts.get("default"));
    }

    public void count(String label) {
        if (counts.get(label) == null) {
            counts.put(label, 1);
        } else {
            counts.replace(label, counts.get(label) + 1);
        }

        System.out.println(label + ": " + counts.get(label));
    }

    public void countReset(String label) {
        counts.replace(label, 0);
    }

    public void log(Object message) {
        System.out.println(Context.toString(message));
    }
}

File: src/main/resources/console.js
function Console() {
    this.counts = {"default": 0};

    this.count = function(label) {
        countToPrint = 1;
        if (counts[label] != null) {
            countToPrint = counts[label];
        }
        java.lang.System.out.println(label + ": " + counts[label]);
    };

    this.countReset = function(label) {
        counts[label] = 0;
    };

    this.log = function(message) {
        java.lang.System.out.println(message);
    };
}

var console = new Console();